spanner/doc.go

```diff
@@ -357,3 +357,7 @@ at https://godoc.org/go.opencensus.io/trace. OpenCensus tracing requires Go 1.8
 or higher.
 */
 package spanner // import "cloud.google.com/go/spanner"
+
+// clientUserAgent identifies the version of this package.
+// It should be the same as https://pkg.go.dev/cloud.google.com/go/spanner.
+const clientUserAgent = "spanner-go/v1.12.0"
```

Comment:
Could this be dynamic? I am afraid we will forget to update this when a new release comes out.

Reply:
For instance, see https://github.com/googleapis/google-cloud-go/commit/4b1db5fe8a5217bf17941627ec16ae8830aa794f.

It would be nice to have a review from @olavloite here too.

Reply:
It seems we can not do it dynamically in Go. For example, see https://github.com/googleapis/google-cloud-go/blob/master/bigtable/doc.go#L121, and https://github.com/googleapis/google-cloud-go/blob/master/internal/version/version.go#L29. And there is an issue about using semver instead of date, see https://github.com/googleapis/google-cloud-go/issues/2749.

Reply:
We should really automate this in some way, otherwise it will certainly be forgotten for future releases. The consts that are mentioned above are also automatically generated as part of the release process. The https://github.com/googleapis/google-cloud-go/blob/master/internal/version/version.go#L29 version is generated by this file: https://github.com/googleapis/google-cloud-go/blob/master/internal/version/update_version.sh

Could we for now at least add a `TODO` here that this needs to be automated with a reference to the bash script mentioned above? This is probably something that we would want to implement for all submodules and integrate in the automated release processes.

Reply:
I agree. Bigtable is the only other client that does this today, and recent PRs have shown that the value has been forgotten about over time. If we were going to start doing something like this I think we should have a consistent header key and value that all of our clients use.

Reply:
I have added a TODO for automating the user agent. There is a plan to use the library version instead of a date, see #2749, so I guess the best solution is to figure out a way to update the